nima_core/cache.py

```
@@ -0,0 +1,254 @@
+        self._cache = LRUEmbeddingCache(capacity=cache_capacity, ttl_seconds=cache_ttl)
+        self._enable_precompute = enable_precompute
+        self._precompute_hits = 0
+        self._cache_hits = 0
+        self._slow_hits = 0
+
+    def warmup(self, texts: List[str]):
+        """Pre-compute embeddings for common queries."""
+        if not self._enable_precompute or not texts:
+            return
+        embeddings = embed_batch(texts)
+        for text, emb in zip(texts, embeddings):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)